### DISCUSSION TITLE
Log flume photo sections use wrong supports

### DISCUSSION THREAD

**Discussion by ZeeMaji** [OpenRCT2/OpenRCT2#21322] [Feature requests]
### Operating System

Linux (OpenSUSE Tumbleweed)

### OpenRCT2 build

OpenRCT2 0.4.6 AppImage Actions Build f773adc508663b06151584bf3ab0c7fab0db84c7

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

The bug also exists in the original game (RCT2/RCTC), This bug is a graphical glitch or error

### Describe the issue

This appears to be a mistake from the original game, the log flume track uses the thick square supports but it's photo section uses the thin supports from the junior/water coasters.

### Steps to reproduce

1. Build a log flume photo section above the ground
2. The supports don't match the supports used by the rest of the track

### Attachments

![Screenshot_select-area_20231010134459](https://github.com/OpenRCT2/OpenRCT2/assets/42477864/9580bf23-fa15-41a1-9556-49ccaba77c03)
![image_original](https://github.com/OpenRCT2/OpenRCT2/assets/42477864/9f9a4d7f-b912-4dd0-99d2-fd3418184d57)

**Comment by ocalhoun6** [comment]
I think this is working as intended? Pretty sure the log flume isn't the only ride in which the on-ride photo section gets unique supports...

And it makes some real-world sense that a photo section might use different supports than the rest of the track.

**Comment by BodocCalado** [comment]
It looks strange and I think there is no obvious reason why to decide against the usual supports for designing this track piece.

**Comment by ZeeMaji** [comment]
> I think this is working as intended? Pretty sure the log flume isn't the only ride in which the on-ride photo section gets unique supports...

I just checked every other ride type with photo sections and none of them have different supports from the main track, so this definitely seems unintended to me.

**Comment by jonniesweb** [comment]
If this is deemed worth changing, the relevant code for the photo section supports looks to be here: https://github.com/OpenRCT2/OpenRCT2/blob/88608e0b7e9c558fbd27ca18a1890a8519b4bf97/src/openrct2/ride/water/LogFlume.cpp#L804-L813

**Comment by ZehMatt** [comment]
Is this really a mistake? Are there any other cases where it would use those alternate supports? If not then its probably intentional. I'm not sure if we should change them for the sake of changing them.

**Comment by ZeeMaji** [comment]
Going by the paint code the only piece with these supports is the photo section, as searching for "fork" in logflume.cpp only shows the photo section. So unless there's something weird it seems the photo section is the only one iwth these supports.

**Comment by ZehMatt** [comment]
> Going by the paint code the only piece with these supports is the photo section, as searching for "fork" in logflume.cpp only shows the photo section. So unless there's something weird it seems the photo section is the only one iwth these supports.

So its probably intentional, if we make it use the standard ones then those sprites will also become unused.

**Comment by ZeeMaji** [comment]
These supports are the same ones used by the junior/water coasters along with some others, they wouldn't become unused.

To me it feels like during the development of RCT2 (as the log flume photo section was not in RCT1) the log flume's photo section paint code was just copy pasted from the junior/water coaster, and chris forgot to change the supports to match the log flume's supports.

**Comment by Gymnasiast** [comment]
> > Going by the paint code the only piece with these supports is the photo section, as searching for "fork" in logflume.cpp only shows the photo section. So unless there's something weird it seems the photo section is the only one iwth these supports.
> 
> So its probably intentional,  if we make it use the standard ones then those sprites will also become unused.

No, both the Junior RC and Water Coaster use these "fork" supports, so they would not become unused. And the Water Coaster, being the only water ride in RCT1 that supported an on-ride photo, is also the most likely place that Chris Sawyer copied the code from.

To me, it seems extremely likely that this is a bug. Most Log Flumes have their photo section directly on the ground, which is how I think this bug slipped through.
